### ISSUE TITLE
Opening an advanced vending machine hard crash

### ISSUE THREAD

**Issue by magicalex234** [Iskallia/Vault-public-S1#106]
It appears that putting multiple traders in an advanced vending machine and then opening it hard crashes your game. The reason i says "appears" is because i'm not 100% positive on the exact conditions that cause it to crash, but I'll tell you what i know:
1. putting a single trader in the advanced vending machine had zero issues.
2. after getting multiple traders in a vault run, i renamed them all different usernames, and put 4 in the vending machine at once. Then upon opening the advanced vending machine, my game froze and then eventually after about a minute, hard crashed by saying "saving world" (as if you logged out), and then when it finished that, crashed to the title screen
3. Logging back in and opening the vending machine causes the exact same crash
4. breaking the vending machine is ok, so i was able to break it, then give myself new traders in creative mode. At the time, i thought it was an issue with one of the trader's name that i put in, so i tried with different names. Still put in 4 and crashed when i opened it. I also tried a 3rd time with different names and got the same error
5. The crash report (which i can maybe link, but would require me to crash my game again) does say the issue came in rendering the advanced vending machine
6. I never opened the advanced vending machine with 2 or 3 traders, so i can't say if that would crash the game
7. I never put multiple traders with the same name in the vending machine, so i can't say if that would crash the game
8. I never tested any of this on a regular vending machine, but i can tell you that putting a single trader in it works fine.
I hope that's enough information for you to try and recreate and pinpoint the bug.

**Comment by jmilthedude** [comment]
Are you able to attach a crash report?

**Comment by magicalex234** [comment]
do you know where to find old crash reports? or should i just recreate the scenario

**Comment by jmilthedude** [comment]
Should be in the instance root folder in “crash-reports”.

**Comment by iep1000** [comment]
I'm also having this issue. I was able to load the traders into the vending machine previously. Then I logged out. When I tried to log back into the game, my character was still standing in front of the vending machine. I tried to click on it and the game crashed. As long as I don't try to access the vending machine the game functions perfectly fine. But when I try to empty it or interact with the traders it crashes.

 I pasted the crash report here. I think I got everything but I'm not sure.

Crash Report UUID: 74ac3d7b-73db-457e-99b8-275ee9ad11ce
	Patchouli open book context: n/a
	Launched Version: MultiMC5
	Backend library: LWJGL version 3.2.2 build 10
	Backend API: AMD Radeon Pro 5500M OpenGL Engine GL version 2.1 ATI-3.10.18, ATI Technologies Inc.
	GL Caps: Using framebuffer using ARB_framebuffer_object extension
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'forge'
	Type: Client (map_client.txt)
	GPU Warnings: version: 2.1
	Graphics mode: fancy
	Resource Packs: vanilla, mod_resources, quark:emote_resources (incompatible), file/VanillaTweaks_r348299.zip
	Current Language: English (US)
	CPU: 16x Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
[23:58:18] [Render thread/INFO] [STDOUT/]: [net.minecraft.util.registry.Bootstrap:func_179870_a:123]: #@!@# Game crashed! Crash report saved to: #@!@# /Applications/MultiMC.app/Contents/MacOS/instances/Vault_Hunters_-_Official_Modpack-1.0/minecraft/crash-reports/crash-2021-04-18_23.58.18-client.txt
AL lib: (EE) alc_cleanup: 1 device not closed
Process exited with code 255.

**Comment by Kennardtalks** [comment]
I also have the same issue, I tried the advanced vending machine twice with only 1 and 2 traders in it, different user names, didn't work. I was able to interact with them in regular vending machines. But if I use the advanced one it's a guaranteed crash whenever you right click on it. 

The only difference its that I started my world in the first release and just updated, If I create a new world it works fine. 

If there's anyway to fix it without loosing my progress I'll be more than glad to now it. You can find the crash report below.

[Vending machine crash-2021-09-22_18.42.54-client.txt](https://github.com/Iskallia/Vault-public/files/7214611/Vending.machine.crash-2021-09-22_18.42.54-client.txt)

Curse Forge overwolf launcher. Modpack was installed directly from the launcher. 

Current Modpack version: Vault Hunters - Official Modpack-1.3-hotfix

World type: default. 

I made the test in survival and creative, its always the same results.

Tysm for your time and hard work.

**Comment by waidesoper** [comment]
Okay, this may sound a little crazy, but are you playing in windowed or full screen?

**Comment by Kennardtalks** [comment]
Always full screen.

**Comment by waidesoper** [comment]
Can you try windowed?

**Comment by Kennardtalks** [comment]
I Just did, and it works! 🥴 
But, why? And does that means that that will be the only way to do it?

**Comment by waidesoper** [comment]
_That's so weird, you might need able to change your UI size and go full screen, some weird combination of your resolution and UI scale were causing the issue. I've never seen this before. Can you try full screen at UI size 2 and see if that works?_

This statement above is wrong. The issue above is caused by having no trades in the trader we believe

**Comment by Kennardtalks** [comment]
Ok, so, that was really weird, but now it works in full screen without further problems, my UI was set to scale 3 but after I tried windowed and went back to full screen it works, I place a new one an it works as well and I closed and reopened the game and still works so I think that solved the Issue. 

Completely rare but TYSM, that actually solve it. :)

**Comment by waidesoper** [comment]
Thanks for the feedback. I'm happy to help

**Comment by ghost** [comment]
can I confirm that you all generated trader core trades, not just renamed them?  like right click to generate trade?

**Comment by Kennardtalks** [comment]
Yes, first right click to generate the trade and then shift R-click to rename.

**Comment by ghost** [comment]
oh, well then I found another bug that produces the same error, go me !  lol
